Modules/Challenges/test/SpecFlowChallenges.Specs/StepDefinitions/ChallengesApiStepDefinitions.cs

```diff
@@ -0,0 +1,219 @@
+    }
+
+    [Given(@"the Accept header is '([^']*)'")]
+    public void GivenTheAcceptHeaderIs(string acceptHeader)
+    {
+        _requestConfiguration.AcceptHeader = acceptHeader;
+    }
+
+    [Given(@"a Challenge c")]
+    public async Task GivenAChallengeC()
+    {
+        using (new AssertionScope())
+        {
+            var challengeResponse = await _challengeApi.CreateChallengeAsync(new RequestConfiguration());
+            challengeResponse.IsSuccessStatusCode.Should().BeTrue();
+
+            if (challengeResponse.Data != null)
+            {
+                _challengeId = challengeResponse.Data?.Result?.Id;
+                _challengeId.Should().NotBeNullOrEmpty(because: "Required value for 'Id' is missing.");
+            }
+            else
+            {
+                challengeResponse.Data.Should().NotBeNull();
+            }
+        }
+    }
+
+    [When(@"a POST request is sent to the Challenges endpoint with")]
+    public async Task WhenAPOSTRequestIsSentToTheChallengesEndpointWith(Table table)
+    {
+        // Check if the request was made by an authenticated user
+        _isAuthenticatedChallenge = !string.IsNullOrEmpty(_requestConfiguration.TokenResponse?.AccessToken);
+
+        var requestConfiguration = table.CreateInstance<RequestConfiguration>();
+        _challengeResponse = await _challengeApi.CreateChallengeAsync(requestConfiguration);
+    }
+
+    [When(@"a POST request is sent to the Challenges endpoint")]
+    public async Task WhenAPOSTRequestIsSentToTheChallengesEndpoint()
+    {
+        // Check if the request was made by an authenticated user
+        _isAuthenticatedChallenge = !string.IsNullOrEmpty(_requestConfiguration.TokenResponse?.AccessToken);
+
+        _challengeResponse = await _challengeApi.CreateChallengeAsync(_requestConfiguration);
+    }
+
+    [When(@"a GET request is sent to the Challenges/\{id} endpoint with Id ""([^""]*)""")]
+    public async Task WhenAGETRequestIsSentToTheChallengesIdEndpointWithId(string id)
+    {
+        switch (id)
+        {
+            case "c.Id":
+                id = _challengeId!;
+                break;
+            case "a valid Id":
+                id = "CHLjVPS6h1082AuBVBaR";
+                break;
+        }
+        _challengeResponse = await _challengeApi.GetChallengeByIdAsync(_requestConfiguration, id);
+    }
+
+    [Then(@"the response content includes an error with the error code ""([^""]*)""")]
+    public void ThenTheResponseContentIncludesAnErrorWithTheErrorCode(string errorCode)
+    {
+        if (_challengeResponse.Content != null)
+        {
+            var errorMessage = JsonConvert.DeserializeObject<ErrorResponse>(_challengeResponse.Content);
+            if (errorMessage != null)
+            {
+                errorMessage.Error.Code.Should().Be(errorCode);
+            }
+            errorMessage.Should().NotBeNull();
+        }
+        _challengeResponse.Content.Should().NotBeNull();
```

Comment:
Again this weird "pattern" in you place an assertion after an if, which both check for the same condition. You can simplify this a lot:

```suggestion
        _challengeResponse.Content.Should().NotBeNull();
        
        var error = JsonConvert.DeserializeObject<ErrorResponse>(_challengeResponse.Content);        
        
        error.Should().NotBeNull();        
        error.Error.Code.Should().Be(errorCode);
```

I've seen this in some other places as well. Please make sure to refactor it everywhere.